app/js/components/orderEntry/addForm/AddForm.jsx

```diff
@@ -217,19 +217,22 @@ export class AddForm extends React.Component {
 
     if (Object.values(this.state.fieldErrors).includes(true)) return true;
 
-    else if (duration && !durationUnit) return true;
-
-    else if ((this.props.careSetting.display === 'Outpatient') &&
+    else if (this.state.formType === 'Standard Dosage') {
+      if (duration && !durationUnit) return true;
```

Comment:
Same here...Basically, we can remove all the switch statement and replace with return statements.